filter_reduce_sort.playground/Contents.swift

```diff
@@ -0,0 +1,132 @@
+//: [⇐ Previous: 07 - Challenge - forEach and map](@previous)
+//: ## Episode 08: filter, reduce, and sort
+
+// --------------------------------------
+let arrayOfDwarfArrays = [
+  ["Sleepy", "Grumpy", "Doc", "Bashful", "Sneezy"],
+  ["Thorin", "Nori", "Bombur"]
+]
+// --------------------------------------
+
+//: filter
+
+let dwarvesAfterM = arrayOfDwarfArrays.flatMap { dwarves -> [String] in
+  var afterM: [String] = []
+  for dwarf in dwarves where dwarf > "M" {
+    afterM.append(dwarf)
+  }
+  return afterM
+}
+
+
+let dwarvesAfterM2 = arrayOfDwarfArrays.flatMap { dwarves -> [String] in
+  dwarves.filter { $0 > "M" }
+}
+
+//: Thomas challenge
+let dwarvesAfterM3 = arrayOfDwarfArrays.flatMap { $0.filter { $0 > "M"} }
```

Comment:
this is still the same version as M2 just written without the `dwarves -> [String] in`

Reply:
Updated!